src/main/cpp/src/iceberg_truncate.hpp

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) 2025, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cudf/column/column.hpp>
+#include <cudf/column/column_view.hpp>
+#include <cudf/utilities/default_stream.hpp>
+#include <cudf/utilities/memory_resource.hpp>
+
+namespace spark_rapids_jni {
+
+/**
+ * @brief Truncate towards negative infinity direction for types: integral or decimal.
+ *
+ * The input type can be INT32, INT64, DECIMAL32, DECIMAL64 or DECIMAL128.
+ * Note: For DECIMAL types, the truncation is performed on the integer representation
+ *
+ * For positive values, Iceberg truncation is: value - (value % width)
+ * For negative values, this uses a floored modulo approach:
+ * value - (((value % width) + width) % width)
+ *
+ * Example:
+ * - truncate_integer(10, 5) = 0
```

Comment:
Is that right?  Per the formula, shouldn't the truncation result be `10`?

Reply:
> Just a moment, are the arguments flipped in the example?
Yes. Updated the comments to make the examples clear.

Reply:
> Is a negative width even valid, by the way?
I tried to find the constraint (width > 0), but not found.
Let's first assume the width must be positive. Negative width is meaningless.